array.c

```diff
@@ -2284,6 +2294,73 @@ ary_aset_by_rb_ary_splice(VALUE ary, long beg, long len, VALUE val)
  *
  *  Assigns elements in +self+; returns the given +object+.
  *
+ *  In brief:
```

Comment:
IMO the "in brief" section stops being useful when it tries to exhaustively cover all the cases. It becomes harder to read than the regular documentation.

Reply:
Agree;  pared.